Userland/Libraries/LibJS/Runtime/Array.cpp

```diff
@@ -65,25 +65,54 @@ Array* Array::typed_this(VM& vm, GlobalObject& global_object)
 
 JS_DEFINE_NATIVE_GETTER(Array::length_getter)
 {
-    auto* array = typed_this(vm, global_object);
-    if (!array)
+    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    if (!this_object) {
         return {};
-    return Value(array->indexed_properties().array_like_size());
+    }
+
+    if (!this_object->is_array()) {
+        if (this_object->has_own_property(vm.names.length)) {
+            Value own_length = this_object->get_own_property(vm.names.length, {});
+            if (vm.exception())
+                return {};
+            return own_length.value_or(js_undefined());
+        }
+    }
+
+    size_t length = this_object->indexed_properties().array_like_size();
+    if (vm.exception())
+        return {};
+    return Value(length);
 }
 
 JS_DEFINE_NATIVE_SETTER(Array::length_setter)
 {
-    auto* array = typed_this(vm, global_object);
-    if (!array)
+    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    if (!this_object)
+        return;
+
+    if (!this_object->is_array()) {
+        this_object->put_own_property(vm.names.length.to_string_or_symbol(), value, default_attributes, PutOwnPropertyMode::Put, true);
+        if (vm.exception())
+            return;
         return;
+    }
+
     auto length = value.to_number(global_object);
     if (vm.exception())
         return;
-    if (length.is_nan() || length.is_infinity() || length.as_double() < 0) {
+
+    u32 val = length.as_double();
+
+    if (val != length.as_double()
+        || length.is_nan()
+        || length.is_infinity()
+        || length.as_double() < 0
+        || length.as_double() > NumericLimits<u32>::max()) {
```

Comment:
Is style like this allowed? `clang-format` is ok with it and the line was quite long.